lib/group_site_manager.py

```diff
@@ -481,6 +484,50 @@ def _upload_dtmd(self, numbers: List[str]):
     finally:
       driver.quit()
 
+  def _login_emb(self) -> WebDriver:
+    group_config = self.config['groups']['embdeals']
+    username = group_config['username']
+    password = group_config['password']
+    driver = self.driver_creator.new()
+
+    # the website is terrible, give it leeway
+    driver.set_page_load_timeout(30)
+    driver.set_script_timeout(30)
+    driver.implicitly_wait(30)
+    self._load_page(driver, EMB_URL)
+    
+    driver.find_element_by_css_selector('input[name="email"]').send_keys(username)
+    driver.find_element_by_css_selector('[name="password"]').send_keys(password)
+    driver.find_element_by_css_selector('app-login .mat-button-wrapper').click()
+    time.sleep(3)
+    return driver
+
+  def _upload_emb(self, numbers) -> None:
+    driver = self._login_emb()
+    try:
+      time.sleep(1)
+      #Open menu and navigate to Tracking page
+      driver.find_element_by_xpath("//*[contains(text(), 'menu')]").click()
+      #EMB site is slow
+      time.sleep(1)
+      driver.find_element_by_xpath("//*[contains(text(), 'Tracking')]").click()
+      time.sleep(1)
+
+      #add trackings
+      driver.find_element_by_xpath("//*[contains(text(), 'Bulk Tracking')]").click()
+      driver.find_element_by_css_selector("[name='numbers']").send_keys('\n'.join(numbers))
+
+      #submit - EMB site is slow
+      driver.implicitly_wait(120)
+      driver.find_element_by_css_selector('.mat-raised-button span.mat-button-wrapper').click()
+
+      #wait for trackings to be saved
+      driver.find_element_by_xpath("//*[contains(text(), 'saved')]")
+      time.sleep(4)
```

Comment:
Is this sleep necessary? We should be done at this point right?

Reply:
Sleep removed